src/core.ts

```diff
@@ -93,26 +92,32 @@ export function addEvent(type: string, eventState: any, time?: number) {
   nextPayloadLength += eventStr.length;
 
   // Edge case:
-  // Don't reschedule upload when next payload consists of exactly one XhrError instrumentation event.
+  // Don't schedule next upload when next payload consists of exactly one XhrError instrumentation event.
   // This helps us avoid the infinite loop in the case when all requests fail (e.g. dropped internet connection)
   // Infinite loop comes from sending instrumentation about failing to deliver previous delivery failure instrumentation.
-  let rescheduleUpload = !(eventState && eventState.type === Instrumentation.XhrError && nextPayload.length === 1);
-  if (rescheduleUpload) {
-    if (timeout) {
-      clearTimeout(timeout);
-    }
+  scheduleUpload = scheduleUpload && !(event.state && event.state.type === Instrumentation.XhrError && nextPayload.length === 1);
```

Comment:
>scheduleUpload [](start = 2, length = 14)

nit: I'm against modifying input parameters like this as a practice and I don't think the byte count savings are worth the readability/maintainability.